src/Symfony/Component/HttpKernel/Fragment/InlineFragmentRenderer.php

```diff
@@ -119,10 +119,14 @@ protected function createSubRequest($uri, Request $request)
         // Sub-request object will point to localhost as client ip and real client ip
         // will be included into trusted header for client ip
         try {
-            if ($trustedHeaderName = Request::getTrustedHeaderName(Request::HEADER_CLIENT_IP, false)) {
+            if (method_exists(Request::class, 'getTrustedHeaderName') && $trustedHeaderName = Request::getTrustedHeaderName(Request::HEADER_CLIENT_IP, false)) {
                 $currentXForwardedFor = $request->headers->get($trustedHeaderName, '');
 
                 $server['HTTP_'.$trustedHeaderName] = ($currentXForwardedFor ? $currentXForwardedFor.', ' : '').$request->getClientIp();
+            } elseif (Request::HEADER_X_FORWARDED_FOR & Request::getTrustedHeaderSet()) {
```

Comment:
Actually, the non-deprecated API should be tried first, to use it when it is available IMO